src/raftstore/store/read_queue.rs

```diff
@@ -123,7 +123,7 @@ impl ReadIndexQueue {
         self.handled_cnt = 0;
     }
 
-    pub fn clear_uncommitted(&mut self, term: u64) {
+    pub fn clear_uncommitted_on_role_change(&mut self, term: u64) {
         let mut removed = 0;
         for mut read in self.reads.drain(self.ready_cnt..) {
             self.contexts.remove(&read.id);
```

Comment:
I think this operation is not necessary anymore since the map is cleared at the next step.